src/libmongoc/src/mongoc/mongoc-database.c

```diff
@@ -1248,41 +1248,70 @@ _mongoc_get_encryptedFields_from_server (mongoc_client_t *client,
    return ret;
 }
 
-mongoc_collection_t *
-mongoc_database_create_collection (mongoc_database_t *database,
-                                   const char *name,
-                                   const bson_t *opts,
-                                   bson_error_t *error)
+bool
+_mongoc_get_collection_encryptedFields (mongoc_client_t *client,
+                                        const char *dbName,
+                                        const char *collName,
+                                        const bson_t *opts,
+                                        bson_t *encryptedFields,
+                                        bson_error_t *error)
 {
-   bson_t encryptedFields = BSON_INITIALIZER;
+   bson_init (encryptedFields); // Initially empty
```

Comment:
Re: initializing the `bson_t`, this is inherently just confusing and somewhat broken. The `storeDocRef` will overwrite the `bson_t` completely without destroying it, so any allocated data will be leaked. The `bson_init` doesn't allocate anything, so this is never a problem. The `bson_init()` also makes it safe for the caller to destroy, even if there is no document written to it. The exitsing `_mongoc_get_encryptedFields_from_map()` already exhibited this behavior.

IMHO, this "maybe it allocates sometimes, but it might also be a view, and the zero-init state is meaningless, and you have to init the object to destroy it, and `bson_destroy` might also free the pointer, but that depends on whether it comes from `bson_new`, and also we'll terminate if you try to modify the object and the pointee happens to be read-only" makes `bson_t` a very bad API. 😃